javascript/first-js-oop-project.md

```diff
@@ -0,0 +1,155 @@
+# Mini Project: Object Oriented Programming in JS
+
+### Projected Time
+
+Total: 65 min - 80min
+- Independent Practice: 15 min
+- Materials: 25 min
+- Lesson: 30 min
+
+### Prerequisites
+
+- [Java script lesson 7](../javascript/javascript-7-oop.md)
+
+
+### Objectives
+
+- To use the new JS object-oriented programming skills.
+- To create a few event objects and push them to an array.
+- Use jQuery to display the list of events to an html page.
+
+### Specific Things To Teach
+
+- Creating event array.
+- Iterating through the array.
+- Displaying array items in HTML page.
+
+### Materials
+
+- [Documentation on each statement of jQuery](http://api.jquery.com/jquery.each/)
+- [Arrays in javaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)
+- [.html function of jQuery](http://api.jquery.com/html/)
+
+### Lesson
+
+- The below example is the just a part of a mini project that will help you to learn basic concept about arrays in javaScript and iterating through the arrays. The below code will allow you to learn about displaying the array objects in the web pages.
+- Once you are done with the code, run the index.html file in your browser to check the result.
+- Keep on refreshing the browser to keep track of the changes you make everytime.
+- Follow the steps described below:
+
+1. Create a js file `event.js`.
+2. Create an empty Event array.
+   ```javascript
+      var event_array = new Array();
+   ```
+3. Create an Event class.
+    ```javascript
+       class Event
+       {
+        constructor(name, description)
+        {
+         this.name = name;
+        this.description = description;
+        }
+       }
+    ```
+4. Now Use to class to create an object and assign values to them.
+   ```javascript
+      // The below statement creates an object.
+      let event_obj1 = new Event("KLOS Golden Gala", "An evening with hollywood vampires");
+   ```
+5. Create few more objects with different value.
+   ```javascript
+    let event_obj2 = new Event("Skillet & Sevendust", "Victorious war tour");
+    let event_obj3 = new Event("Jenny Lewis", "On the line tour 2019");
+   ```
+6. Now push the objects created into the array. In order to push objects in the array, do the following.
+   ```javascript
+      // pushing single object to an array
+      event_array.push(event_obj1);
+      // pushing multiple objects to an array at once
+      event_array.push(event_obj1, event_obj2, event_obj3);
+   ```
+7. Keep checking everytime you push the element using ***console.log***.
+   ```javascript
+       //inorder to check whether the elements are pushed, Use Console.log
+       console.log(event_array);
+   ```
+
+8. After that you have created the array now we need to write the jQuery code to iterate through it in the same js file event.js .
```

Comment:
Does it make sense to introduce query and then show them html?

Reply:
There is nothing much in the HTML file, the most important part was jquery, so I did it that way.
Do you want me to put the HTML file above all?